src/backend/common/queries/database_query.py

```diff
@@ -29,3 +30,15 @@ def fetch_async(self) -> TypedFuture[QueryReturn]:
             # Type-hinting the tasklet decorator is hard, but it consumes
             # the generator and returns the overall future
             return query_result  # pyre-ignore
+
+    def fetch_dict(self, version: int) -> dict:
+        return self.fetch_dict_async(version).get_result()
+
+    @ndb.tasklet
+    def fetch_dict_async(self, version: int) -> TypedFuture[dict]:
+        query_result = yield self.fetch_async()
+        if query_result is None:
+            raise DoesNotExistException()
+        # Type-hinting the tasklet decorator is hard, but it consumes
+        # the generator and returns the overall future
+        return self.DICT_CONVERTER(query_result).convert(version)  # pyre-ignore
```

Comment:
is this ignore needed? We can look into better typing here later if it is...

I think we need to make some improvements to better handle the generators here, but I'm not quite sure what yet